packages/hydrogen/src/seo/generate-seo-tags.ts

```diff
@@ -3,9 +3,8 @@ import type {WithContext, Thing} from 'schema-dts';
 import type {ComponentPropsWithoutRef} from 'react';
 
 const ERROR_PREFIX = 'Error in SEO input: ';
-// TODO: Refactor this into more reusible validators
-// or use a library like zod to do this if we decide
-// to use it in other places. @cartogram
+// TODO: Refactor this into more reusible validators or use a library like zod to do this if we decide to use it in
```

Comment:
I think we should create an issue or a discussion about `zod` to get it going. 

I think there's enough use cases for it (analytics, seo, cart) etc